pwndbg/elf.py

```diff
@@ -120,28 +122,55 @@ def reset_ehdr_type_loaded():
     global ehdr_type_loaded
     ehdr_type_loaded = 0
 
+@pwndbg.abi.LinuxOnly()
+def find_elf_magic(pointer, max_pages=1024, search_down=False):
+    """Find the page address which contains the ELF headers
+    by comparing the ELF magic with first 4 bytes.
+
+    Returns:
+        An integer address of ELF page base
+        None if not found within the page limit
+    """
+    base = pwndbg.memory.page_align(pointer)
+    step = pwndbg.memory.PAGE_SIZE
+    if not search_down:
+        step = -step
```

Comment:
Shouldn't it be `if search_down: step = -step`? Isn't searching down a searching over lower addresses?

Reply:
The first image in my brain is a stack view and the lower address is at the top, so when I thing the "down" is the higher address, lol. I think I get it the "down" means like the "count down" which is decreasing. I will update later.